File: server/src/main/java/com/viewfromaside/apdo/model/Note.java
package com.viewfromaside.apdo.model;

import com.viewfromaside.apdo.enums.NoteVisibility;

public class Note extends BaseModel {
    private String title;
    private String content;
    private NoteVisibility visibility;

    public Note() {

    }

    public Note(String id, String title, String content, NoteVisibility visibility) {
        super();
        this.id = id;
        this.title = title;
        this.content = content;
        this.visibility = visibility;
    }

    public String getTitle() {
        return this.title;
    }
    
    public String getContent() {
        return this.content;
    }

    public NoteVisibility getVisibility() {
        return this.visibility;
    }
}

File: server/models/note.go
package models

import (
	"time"

	"github.com/apdo/server/utils"
)

type NoteVisibility string

const (
	PUBLIC  NoteVisibility = "public"
	PRIVATE NoteVisibility = "private"
)

type Note struct {
	BaseModel
	Title      string         `json:"title"`
	Content    string         `json:"content"`
	Visibility NoteVisibility `json:"visibility"`
}

func NewNote() Note {
	now := time.Now()
	return Note{
		BaseModel: BaseModel{
			ID:        utils.GenerateID(7),
			CreatedAt: now,
			UpdatedAt: now,
			CreatedBy: "system",
		},
		Title:      "Default Title",
		Content:    "Default Content",
		Visibility: PUBLIC,
	}
}
